### PR TITLE
Update AWS OIDC docs

### PR DESCRIPTION
**Pull request by hnrkndrssn**
Try and make it easier for users to fall in the pit of success.

Updates based on feedback from a customer trying to get AWS OIDC setup

[sc-69507]

### CODE REVIEW

File: src/pages/docs/infrastructure/accounts/aws/index.md [hunk 1]
```diff
@@ -33,18 +33,35 @@ See the [AWS documentation](https://oc.to/aws-access-keys) for instructions to c
...
-1. Provide a description for the account.
-1. Set the Role ARN to the ARN from the identity provider associated role.
-1. Set the Session Duration to the Maximum session duration from the role, in seconds.
-1. Click the **SAVE AND TEST** to save the account and verify the credentials are valid.
+2. Add a memorable name for the account.
+3. Provide a description for the account.
+4. Set the **Role ARN** to the ARN from the identity provider associated role.
+5. Set the **Session Duration** to the Maximum session duration from the role, in seconds.
+6. Click **SAVE** to save the account.
+7. Before you can test the account you need to add a condition to the identity provider in AWS:
```
Comment (IsaacCalligeros95) [inline src/pages/docs/infrastructure/accounts/aws/index.md:55]: This is in the _IAM -> Roles -> Trust Relationships_ right? I think spelling that out here is worthwhile, people might just be looking under the Identityprovider

Comment (IsaacCalligeros95) [reply src/pages/docs/infrastructure/accounts/aws/index.md:55]: Should we also include the Principal to better understand the whole trust relationship JSON? 

I'm also not sure if it's worth mentioning that they can avoid the subject requirement, from a security perspective probably better to leave that out but during testing I configured mine on just the audience and that was pretty helpful in my case.

`
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "Federated": "arn:aws:iam::720766170633:oidc-provider/a333-2001-8003-ac03-7500-6dfc-aa25-5c69-abec.ngrok-free.app"
            },
            "Action": "sts:AssumeRoleWithWebIdentity",
            "Condition": {
                "StringEquals": {
                    "a333-2001-8003-ac03-7500-6dfc-aa25-5c69-abec.ngrok-free.app:aud": "a333-2001-8003-ac03-7500-6dfc-aa25-5c69-abec.ngrok-free.app"
                }
            }
        }
    ]
}
`

**Review by IsaacCalligeros95 — COMMENTED** [review]
A couple of minor comments, I'm happy to approve it as is if you don't think they're worth changing

File: src/pages/docs/infrastructure/accounts/aws/index.md (hunk 1 above)
Comment (hnrkndrssn) [reply src/pages/docs/infrastructure/accounts/aws/index.md:55]: > not sure if it's worth mentioning that they can avoid the subject requirement
I'd rather not, the chance that someone then forgets to update it to after testing and leaving their identity provider wide open is higher than I'd like compared to the convenience.

**Review by IsaacCalligeros95 — APPROVED** [review]
LGTM

**Pull request merged**